### PR TITLE
include discover hosts with nmap and API nmap page

### PR DESCRIPTION
**Pull request by jmariomejiap**
## Discover -> tidal tools

image

## API usage

image

### CODE REVIEW

File: pages/user guides/discover_nmap.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+keywords: nmap, ip, address, api, services
+last_updated: September 2020
+summary: "How to use nmap to scan your network"
+sidebar: main_sidebar
+permalink: discover-nmap.html
+---
+
+
+
+As part of your cloud migration journey it is important to have all the tools at your disposal. In addition to our proprietary tools, you can now use [Nmap](https://nmap.org/) to facilitate your discovery process.
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:12]:
```suggestion
As part of your cloud migration journey it is important to have all the tools at your disposal. In addition to [the other discovery methods](https://guides.tidalmg.com), you can use [Nmap](https://nmap.org/) to facilitate a part of the discovery process. 
```

File: pages/user guides/discover_nmap.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+last_updated: September 2020
+summary: "How to use nmap to scan your network"
+sidebar: main_sidebar
+permalink: discover-nmap.html
+---
+
+
+
+As part of your cloud migration journey it is important to have all the tools at your disposal. In addition to our proprietary tools, you can now use [Nmap](https://nmap.org/) to facilitate your discovery process. 
+The way it works is very simple and with the full power of nmap available to you, discovery of even those hard to reach places is possible.
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:13]:
```suggestion
The way it works is very simple and with the full power of Nmap available to you, discovery of even those hard to reach places is possible. 
```

File: pages/user guides/discover_nmap.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+sidebar: main_sidebar
+permalink: discover-nmap.html
+---
+
+
+
+As part of your cloud migration journey it is important to have all the tools at your disposal. In addition to our proprietary tools, you can now use [Nmap](https://nmap.org/) to facilitate your discovery process. 
+The way it works is very simple and with the full power of nmap available to you, discovery of even those hard to reach places is possible. 
+ 
+Nmap has developed some powerful capabilities and good [documentation](https://nmap.org/book/host-discovery-find-ips.html) that will allow you to identify  all the hosts you have running on your network, the ports and services that they have exposed and even some vulnerabilities discovered by the open source community.
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:15]:
```suggestion
Nmap has developed some powerful capabilities and good [documentation](https://nmap.org/book/host-discovery-find-ips.html) that will allow you to identify all the hosts you have running on your network, the ports and services that they have open and running. 
```

File: pages/user guides/discover_nmap.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+---
+
+
+
+As part of your cloud migration journey it is important to have all the tools at your disposal. In addition to our proprietary tools, you can now use [Nmap](https://nmap.org/) to facilitate your discovery process. 
+The way it works is very simple and with the full power of nmap available to you, discovery of even those hard to reach places is possible. 
+ 
+Nmap has developed some powerful capabilities and good [documentation](https://nmap.org/book/host-discovery-find-ips.html) that will allow you to identify  all the hosts you have running on your network, the ports and services that they have exposed and even some vulnerabilities discovered by the open source community. 
+ 
+Nmap is capable of producing its output in an XML file, allowing you to inspect the raw scan output before sending it to the Tidal Migrations API with our CLI,  [Tidal Tools](https://get.tidal.sh). Once uploaded to the Tidal Migrations Platform, you will be able to visualize your network devices, track your complete server inventory and build on this data with other discovery methods, allowing you to make informed decisions on your cloud migration path.
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:17]:
```suggestion
Nmap is capable of producing its output in an XML file, allowing you to inspect the raw scan output before sending it to the Tidal Migrations API with [Tidal Tools](https://get.tidal.sh). Once uploaded to the Tidal Migrations Platform, you will be able to visualize your network devices, track your complete server inventory and build on this data with other discovery methods, allowing you to make informed decisions on your cloud migration path.
```

File: pages/user guides/discover_nmap.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+## Using Nmap with Tidal Tools
+ 
+By leveraging the power of **Tidal Tools** you can send the output generated by Nmap to Tidal migration API.
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:22]:
```suggestion
By leveraging the power of **Tidal Tools** you can send the output generated by Nmap to your Tidal Migrations account.
```

File: pages/user guides/discover_nmap.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+ 
+To get the process started your must follow these steps first:
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:24]:
```suggestion
```

(seems redundant, and it's more than getting started, it's how to do it end to end.)

File: pages/user guides/discover_nmap.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+1. Install [Tidal Tools](https://guides.tidalmg.com/tidal-tools.html)
+2. [Login to Tidal Tools](https://guides.tidalmg.com/tidal-tools.html#login) with **tidal login**
+3. Run Nmap [how to install](#installing-nmap) with the flags of your choosing and save the output on an XML file. For example,
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:27]:
```suggestion
3. Run Nmap ([Need to install Nmap?](#installing-nmap)) with the flags of your choosing and save the output to an XML file. For example, 
```

File: pages/user guides/discover_nmap.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+By leveraging the power of **Tidal Tools** you can send the output generated by Nmap to Tidal migration API.
+ 
+To get the process started your must follow these steps first:
+1. Install [Tidal Tools](https://guides.tidalmg.com/tidal-tools.html)
+2. [Login to Tidal Tools](https://guides.tidalmg.com/tidal-tools.html#login) with **tidal login**
+3. Run Nmap [how to install](#installing-nmap) with the flags of your choosing and save the output on an XML file. For example, 
+```
+sudo nmap -sS -p80,443,8080,8443,1433,1521,27017 <ip-address> -oX my-network.xml
+```
+Note: the **-sS** flag will enable TCP SYN port scan and the **-oX** specifies the output as a XML file.
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:31]: maybe an extra note here:

```
Note: Be sure to replace <ip-address> with the CIDR range for the network you would like to scan, ex. `10.0.0.0/24`
```

File: pages/user guides/discover_nmap.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+[Want more scanning options?](#nmap-usage) 
+ 
+4. Run this Tidal Tools command to upload your previously generated Nmap output to your Tidal Migrations account 
+```
+tidal sync nmap <filename.XML> 
+``` 
+ 
+5. Head out to your Tidal account!
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:39]:
```suggestion
5. Head over to your Tidal Migrations account!
```

Include a link or menu item to navigate too, or screenshot of where to continue the process (once that is in place).

File: pages/user guides/discover_nmap.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+Nmap ("Network Mapper") is a free and open source utility for network discovery and security auditing. Many systems and network administrators also find it useful for tasks such as network inventory, managing service upgrade schedules, and monitoring host or service uptime.
+ 
+You can find all the documentation in its official site [Nmap](https://nmap.org/) and instructions for how to download based on your environment [download](https://nmap.org/download.html)
+
+
+## Nmap usage
+
+Nmap offers a wide range of utilities and commands, such as Port scanning, Host discovery, Service and version detection to name a few. Here some basic examples for how to do some nmap scanning.
+
```
Comment (prees1) [inline pages/user guides/discover_nmap.md:53]: Consider a note here listing the features or attributes that Tidal currently captures. Something like

```
Note: Currently, Tidal Migrations supports the collection of Hosts (IP addresses), PTR records, their open ports, the ports status, the port protocol (TCP/UDP), and the services running in the port, including the version.
```

Thinking so it's clear when someone provides an Nmap file with _more_ data and wondering why its not in the Platform after uploading. We might want to communicate this in other places. Maybe higher up on this page. Or in Tidal Tools itself as well, either in the usage text or the output shown to a user while or after uploading the data.

**Review by prees1 — CHANGES_REQUESTED** [review]
Awesome pair of guides! I think it really covers the use cases well.

A few comments on things to consider or adjust. Some are typos that should definitely be corrected pre-merge.

**Comment by prees1** [comment]
Also, one thing to consider for styling is the `tip` and `callout` styles used elsewhere. They might work well here for some of the `Notes` or `NB` items.

See here for an example - https://github.com/tidalmigrations/tidalmigrations.github.io/blame/master/pages/tidal%20tools/analyze.md#L42

The look like this:
https://guides.tidalmg.com/analyze.html#uploading-to-tidal-migrations-api
image

File: pages/user guides/discover_nmap.md (hunk 9 above)
Comment (jmariomejiap) [reply pages/user guides/discover_nmap.md:38]: great idea!

File: pages/api/Using_with_nmap.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+       "ip_address" : "32.199.2.33",
+	      “ptr_record”: “abc-32-199-2-33.cloudfront.net”,
+        "ports": [
+          {
+            "port": 22,
+            "protocol": "tcp",
+            "status": "open",
+            "service": {
+              "name": "openssh"          
+              “CVES”: {
```
[the remark this answers is not in the document]
Comment (jmariomejiap) [reply pages/api/Using_with_nmap.md:40]: good point, thanks 👍

**Review by prees1 — APPROVED** [review]
looks good!

**Pull request merged**